mapactionpy_arcmap/MapChef.py

```diff
@@ -47,6 +46,9 @@ def removeLayers(self):
                 arcpy.mapping.RemoveLayer(df, lyr)
         self.mxd.save()
 
+    # APS 06/09/2019 The `cook()` method is much too long and has too many nested `if`s and `for`s. Please split this
+    # into the several separate private methods for handling the different scenarios, data types, error conditions 
+    # etc. This refactoring would benefit from adding some unittests surrounding it.
```

Comment:
The `cook()` method is much too long and has too many nested `if`s and `for`s. Please split this into the several separate private methods for handling the different scenarios, data types, error conditions etc. This refactoring would benefit from adding some unittests surrounding it.